File: examples/lib/src/test/java/com/github/neboskreb/log4j2/examples/lib/AwesomeWorkerTest.java
package com.github.neboskreb.log4j2.examples.lib;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class AwesomeWorkerTest {
    @Test
    void doGreatJob() {
        // GIVEN
        AwesomeWorker awesomeClass = new AwesomeWorker();

        // WHEN
        String result = awesomeClass.doGreatJob("foo", "bar");

        // THEN
        assertEquals("foobar", result);
    }

    @Test
    void doNotSoGreatJob() {
        // GIVEN
        AwesomeWorker awesomeClass = new AwesomeWorker();

        // WHEN
        awesomeClass.doNotSoGreatJob();

        // THEN no exceptions thrown
    }
}

File: examples/lib/src/test/kotlin/com/github/neboskreb/log4j2/examples/lib/AwesomeWorkerTest.kt
package com.github.neboskreb.log4j2.examples.lib

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

internal class AwesomeWorkerTest {
    @Test
    fun doGreatJob() {
        // GIVEN
        val awesomeClass = AwesomeWorker()

        // WHEN
        val result = awesomeClass.doGreatJob("foo", "bar")

        // THEN
        Assertions.assertEquals("foobar", result)
    }

    @Test
    fun doNotSoGreatJob() {
        // GIVEN
        val awesomeClass = AwesomeWorker()

        // WHEN
        awesomeClass.doNotSoGreatJob()

        // THEN no exceptions thrown
    }

    @Test
    fun staticLogging() {
        // WHEN
        logSomething()

        // THEN see logs
    }
}
